File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums)
    {
        int n = nums.size();
        int x = (n / 2);
        unordered_map<int, int>mp;

        for(auto &it: nums)
        {
            mp[it]++;
        }

        for(auto &it: mp)
        {
            if(it.second > x)
            {
                return it.first;
            }
        }

        return 0;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        mp = defaultdict(int)
        x = 0
        if n & 1:
            x = (n // 2) + 1
        else:
            x = (n // 2)

        nums.sort()
        ans = []
        
        for i in nums:
            mp[i] += 1

        for key, value in mp.items():
            if value >= x:
                return key

        return 0
